includes/rest-api/class-wp-rest-stories-controller.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * Class AMP_REST_Stories_Controller
+ */
+class AMP_REST_Stories_Controller extends WP_REST_Posts_Controller {
+	/**
+	 * Prepares a single story for create or update. Add post_content_filtered field to save/insert.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return stdClass|WP_Error Post object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_story = parent::prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared_story ) ) {
+			return $prepared_story;
+		}
+
+		if ( isset( $request['content_filtered'] ) ) {
```

Comment:
Shouldn't this field be marked as `required`? It shouldn't be allowed to update the `content` without also updating the underlying corresponding story data. I don't think this `if` condition should be here then.

Reply:
Why does it need to be required?

Reply:
Well, it should at least be `required` if the `content` is also supplied. We would want to force the client to ensure when updating a story that they provide both the rendered AMP story HTML as well as the underlying data that was used to render it. This is my understanding of how the editor will work with the data.

The same should be true vice-versa: if the story data is supplied then the `content` must should also be supplied.

Reply:
I mean if we are going to start renaming fields, should we rename all the fields in the REST API to what they are in our new editor? Either we go with WP naming or we don't here. Some fields is really confusing.

Reply:
What does this have to do with _naming_? 🤔 I thought this is just about making sure required data is supplied.

Reply:
@swissspidy This comment was designed to go with https://github.com/ampproject/amp-wp/pull/3871#discussion_r357643691 . Something must have gone wrong and it was posted here.

Reply:
@westonruter I'll be addressing it being required if `content` is supplied within #3838. Will mark this resolved for the purposes of this PR.